lib/PushManager.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Created by noam.arad on 10/23/2017.
+ */
+
+const io = require('socket.io-client');
+const util = require('util');
+const BackendClient = require('./BackendClientFactory').getBackendClient();
+const logger = require('../common/logger').getLogger('PushManager');
+const kalturaTypes = require('./kaltura-client-lib/KalturaTypes');
+const playListGenerator = require('./playlistGenerator/PlaylistGenerator');
+const RecordingManager = require('./recording/RecordingManager');
+
+let instance = null;
+
+class PushManager {
+
+    constructor()
+    {
+        if (!instance) {
+            instance = this;
+            instance.registeredEntries = {};
+        }
+        return instance;
+    }
+
+    registerEntry(entryId, partnerId, templateName, callback)
+    {
+        const This = this;
+        if (!This.registeredEntries[entryId] || !This.registeredEntries[entryId].templates[templateName].init)
+            BackendClient.registerToPush(templateName, partnerId, entryId).then((registerResult) =>{
+                logger.debug(`registered successfully to push event notifications data [${util.inspect(registerResult)}]`);
+                if (!This.socket)
+                    This._initSocket(registerResult.url, callback);
```

Comment:
the parameter callback of  registerEntry   is not called if the socket already exists...
(since you call it only to the first)